api/receiver/handlers.go

```diff
@@ -33,3 +35,34 @@ func apiResultsCreateHandler(w http.ResponseWriter, r *http.Request) {
 	s := checks.NewAPI(ctx)
 	HandleResultsCreate(a, s, w, r)
 }
+
+func apiPendingTestRunUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
```

Comment:
Not clear cut here, but this feels more like a `PATCH`, since the body provided from the Python code below only has id/status fields (and doesn't modify the other fields?)